components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/par/OAuth2ParEndpoint.java

```diff
@@ -0,0 +1,149 @@
+                        MultivaluedMap<String, String> params) {
+
+        try {
+            handleValidation(request, params);
+            Map<String, String> parameters = transformParams(params);
+            ParAuthResponseData parAuthResponseData =
+                    EndpointUtil.getParAuthService().generateParAuthResponse(parameters);
+            return createAuthResponse(response, parAuthResponseData);
+        } catch (ParClientException e) {
+            return handleParClientException(e);
+        } catch (ParCoreException e) {
+            return handleParCoreException(e);
+        }
+    }
+
+    private Map<String, String> transformParams(MultivaluedMap<String, String> params) {
+
+        Map<String, String> parameters = new HashMap<>();
+        for (Map.Entry<String, List<String>> entry : params.entrySet()) {
+            String key = entry.getKey();
+            List<String> values = entry.getValue();
+            if (!values.isEmpty()) {
+                String value = values.get(0);
+                parameters.put(key, value);
+            }
+        }
+
+        return parameters;
+    }
+
+    private Response createAuthResponse(HttpServletResponse response, ParAuthResponseData parAuthResponseData) {
+
+        response.setContentType(MediaType.APPLICATION_JSON);
+        net.minidev.json.JSONObject parAuthResponse = new net.minidev.json.JSONObject();
+        parAuthResponse.put(OAuthConstants.OAuth20Params.REQUEST_URI,
+                ParConstants.REQUEST_URI_HEAD + parAuthResponseData.getReqUriUUID());
+        parAuthResponse.put(ParConstants.EXPIRES_IN, parAuthResponseData.getExpiryTime());
+        Response.ResponseBuilder responseBuilder = Response.status(HttpServletResponse.SC_CREATED);
+        return responseBuilder.entity(parAuthResponse.toString()).build();
+    }
+
+    private Response handleParClientException(ParClientException exception) {
+
+        String errorCode = exception.getErrorCode();
+        JSONObject parErrorResponse = new JSONObject();
+        parErrorResponse.put(OAuthConstants.OAUTH_ERROR, errorCode);
+        parErrorResponse.put(OAuthConstants.OAUTH_ERROR_DESCRIPTION, exception.getMessage());
+
+        Response.ResponseBuilder responseBuilder;
+        if (OAuth2ErrorCodes.INVALID_CLIENT.equals(errorCode)) {
+            responseBuilder = Response.status(HttpServletResponse.SC_UNAUTHORIZED);
+        } else {
+            responseBuilder = Response.status(HttpServletResponse.SC_BAD_REQUEST);
+        }
+        return responseBuilder.entity(parErrorResponse.toString()).build();
+    }
+
+    private Response handleParCoreException(ParCoreException parCoreException) {
+
+        JSONObject parErrorResponse = new JSONObject();
+        parErrorResponse.put(OAuthConstants.OAUTH_ERROR, OAuth2ErrorCodes.SERVER_ERROR);
+        parErrorResponse.put(OAuthConstants.OAUTH_ERROR_DESCRIPTION, parCoreException.getMessage());
```

Comment:
We shouldn't send server exceptions out directly like this as it could expose internal details. For server errors we should have standard error message that we send out